### DISCUSSION TITLE
complain if custom canister candid file is specified but doesn't exist

### DISCUSSION THREAD

**Discussion by roman-kashitsyn** [dfinity/sdk#2636] [Maintenance Tasks]
If "candid" field in a custom canister definition is set but contains a path to a non-existent file, `dfx` doesn't report and error.
```json
     {
       "canisters": {
         "ledger": {
           "type": "custom",
           "wasm": "ledger.wasm",
           "candid": "non-existent.did"
         }
       }
     }
```
Instead, `dfx` proceeds with the call, which might fail because if incorrect message encoding, e.g.,
```sh
$ dfx canister call ledger account_balance '(record { account = vec{88;11;251;247;125;228;16;248;39;72;186;135;209;137;26;57;209;212;2;65;11;35;125;98;231;139;40;7;77;27;179;129}})'

The Replica returned an error: code 5, message: "Canister rrkah-fqaaa-aaaaa-aaaaq-cai trapped explicitly: Panicked at 'Deserialization Failed: "Fail to decode argument 0 from table0 to record { account : vec nat8 }"', /ic/rs/rust_canisters/dfn_core/src/endpoint.rs:34:41"
```

This makes debugging much harder that it needs to be.

```
dfx --version
dfx 0.8.3
```

**Comment by chenyan-dfinity** [comment]
There used to be a warning when the did file is broken or missing. But then all calls to the main net will have this warning because there is no way to fetch the did file on-chain. Hopefully, this can all be fixed once the custom section is live in replica.
